cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -11,36 +11,100 @@
 
-        this.diskMiB = bc != null && bc.getDisk() != null ? getSizeInMiB(bc.getDisk()) : generateDiskCapacity(spec.getKafka().getStorage());
+        this.replicas = spec.getKafka().getReplicas();
+        this.storage = storage;
+
+        this.diskMiB = bc != null && bc.getDisk() != null ? getSizeInMiB(bc.getDisk()) : generateDiskCapacity(storage);
         this.cpuUtilization = bc != null && bc.getCpuUtilization() != null ? bc.getCpuUtilization() : DEFAULT_BROKER_CPU_UTILIZATION_CAPACITY;
         this.inboundNetworkKiBPerSecond = bc != null && bc.getInboundNetwork() != null ? getThroughputInKiB(bc.getInboundNetwork()) : DEFAULT_BROKER_INBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
         this.outboundNetworkKiBPerSecond = bc != null && bc.getOutboundNetwork() != null ? getThroughputInKiB(bc.getOutboundNetwork()) : DEFAULT_BROKER_OUTBOUND_NETWORK_KIB_PER_SECOND_CAPACITY;
     }
 
     /**
-     * Generate disk capacity configuration from the supplied storage configuration
+     * Generate broker capacity entry for capacity configuration.
+     *
+     * @param brokerId Id of broker
+     * @param diskCapacity Disk capacity configuration
+     * @param doc Documentation for broker entry
+     * @return Broker entry as a JsonObject
+     */
+    private JsonObject generateBrokerCapacity(int brokerId, JsonObject diskCapacity, String doc) {
+        JsonObject brokerCapacity = new JsonObject()
+            .put("brokerId", brokerId)
+            .put("capacity", new JsonObject()
+                .put("DISK", diskCapacity.getValue("DISK"))
+                .put("CPU", Integer.toString(cpuUtilization))
+                .put("NW_IN", Double.toString(inboundNetworkKiBPerSecond))
+                .put("NW_OUT", Double.toString(outboundNetworkKiBPerSecond))
+            )
+            .put("doc", doc);
+        return brokerCapacity;
+    }
+
+    /**
+     * Generate JBOD disk capacity configuration for a broker using the supplied storage configuration
      *
      * @param storage Storage configuration for Kafka cluster
-     * @return Disk capacity configuration value as a Double
+     * @param idx Index of the broker
+     * @return Disk capacity configuration value as a JsonObject for broker idx
+     */
+    private JsonObject generateJbodDiskCapacity(Storage storage, int idx) {
+        JsonObject json = new JsonObject();
+        String name = "";
+        String path = "";
+        String size = "";
+
+        for (SingleVolumeStorage volume : ((JbodStorage) storage).getVolumes()) {
+            name = VolumeUtils.getVolumePrefix(volume.getId());
+            path = AbstractModel.KAFKA_MOUNT_PATH + "/" + name + "/" + AbstractModel.KAFKA_LOG_DIR + idx;
+
+            if (volume instanceof PersistentClaimStorage) {
+                size =  ((PersistentClaimStorage) volume).getSize();
+            } else if (volume instanceof EphemeralStorage) {
+                size = ((EphemeralStorage) volume).getSizeLimit();
```

Comment:
What if the size limit is null? It is not required value.

Reply:
Not sure if I tried this, let me get back to you on this

Reply:
I have  added a conditional to set the size to the `DEFAULT_BROKER_DISK_MIB_CAPACITY` when the provided size limit is null.